spec/models/manager_refresh/inventory_collection_default_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+describe ManagerRefresh::InventoryCollectionDefault do
+  before { allow(MiqServer).to receive(:my_zone) { 'My zone' } }
+  let(:zone) { Zone.first }
+
+  context 'Amazon' do
+    before do
+      ems = new_amazon_ems
+      with_vcr_data { expect { EmsRefresh.refresh(ems) }.to change { VmOrTemplate.count }.from(0) }
+      expect { ems.destroy }.to_not change { VmOrTemplate.count }
+      expect(ExtManagementSystem.count).to be_zero
+    end
+
+    it 'pickups existing Vm records on EMS re-adding' do
+      expect(Vm.count).to_not be_zero
+      with_vcr_data { expect { EmsRefresh.refresh(new_amazon_ems) }.to_not change { Vm.count } }
+    end
+
+    it 'pickups existing Template records on EMS re-adding' do
+      expect(MiqTemplate.count).to_not be_zero
+      with_vcr_data { expect { EmsRefresh.refresh(new_amazon_ems) }.to_not change { MiqTemplate.count } }
```

Comment:
we should also test the :ems_id is set to the new_amazon_ems.id, in both cases (that is the actual reconnect)

Reply:
done